command/agentproxyshared/util/util.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenBao Contributors
```

Comment:
@Huy-Dinh-Reply We already have `github.com/cenkalti/backoff/v4` as a dependency, which is fairly comprehensive. Do we want to use that one, instead?